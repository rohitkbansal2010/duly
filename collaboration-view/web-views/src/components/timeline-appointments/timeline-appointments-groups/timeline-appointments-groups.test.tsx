import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PractitionerRole, PatientAppointmentStatus, PatientAppointmentAlias } from '@enums';
import { calendarTwoBlueIcon } from '@icons';

import {
  TimelineApppointmentsGroups,
  TimelineApppointmentsGroupsPropsType
} from './timeline-appointments-groups';

const mockProps: TimelineApppointmentsGroupsPropsType = {
  icon: calendarTwoBlueIcon,
  backgroundIcon: 'Blue',
  backgroundCount: 'DarkBlue',
  alias: PatientAppointmentAlias.RECENT,
  groups: [
    {
      title: 'Physical Therapy',
      nearestAppointmentDate: '2022-01-10',
      farthestAppointmentDate: '2022-01-29',
      nearestAppointmentPractitioner: {
        id: '1',
        humanName: {
          familyName: 'Ling',
          givenNames: [],
          prefixes: [ 'Dr.' ],
        },
        role: PractitionerRole.MA,
      },
      appointments: [
        {
          startTime: 'Fri, Nov 2, 2021',
          practitioner: {
            id: '1',
            humanName: {
              familyName: 'Ling',
              givenNames: [],
              prefixes: [ 'Dr.' ],
            },
            role: PractitionerRole.MA,
          },
          reason: 'Patient fell - Hip problems',
          minutesDuration: 30,
          status: PatientAppointmentStatus.COMPLETED,
        },
      ],
    },
  ],
};

describe('TimelineApppointmentsGroups', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(<TimelineApppointmentsGroups {...mockProps} />);
  });

  it('should render the TimelineApppointmentsGroups component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it(`should render ${mockProps.groups.length} the TimelineApppointmentsGroup components`, () => {
    const appointmentsGroupsComponents = component.find('TimelineApppointmentsGroup');

    expect(appointmentsGroupsComponents).toHaveLength(mockProps.groups.length);
    appointmentsGroupsComponents.forEach((appointmentsGroupComponent, index) => {
      const expectedProps = {
        eventKey: String(index),
        groupId: `${mockProps.alias}-${index}`,
        icon: mockProps.icon,
        backgroundIcon: mockProps.backgroundIcon,
        backgroundCount: mockProps.backgroundCount,
        ...mockProps.groups[index],
      };

      expect(appointmentsGroupComponent.props()).toEqual(expectedProps);
    });
  });
});
