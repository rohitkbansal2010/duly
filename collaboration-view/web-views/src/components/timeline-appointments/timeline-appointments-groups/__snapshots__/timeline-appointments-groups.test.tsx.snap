// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineApppointmentsGroups should render the TimelineApppointmentsGroups component 1`] = `
<FadedScroll
  height="100%"
>
  <TimelineApppointmentsGroup
    appointments={
      Array [
        Object {
          "minutesDuration": 30,
          "practitioner": Object {
            "humanName": Object {
              "familyName": "Ling",
              "givenNames": Array [],
              "prefixes": Array [
                "Dr.",
              ],
            },
            "id": "1",
            "role": "MA",
          },
          "reason": "Patient fell - Hip problems",
          "startTime": "Fri, Nov 2, 2021",
          "status": "Completed",
        },
      ]
    }
    backgroundCount="DarkBlue"
    backgroundIcon="Blue"
    eventKey="0"
    farthestAppointmentDate="2022-01-29"
    groupId="recentAppointments-0"
    icon="calendar-two-blue.svg"
    key="0"
    nearestAppointmentDate="2022-01-10"
    nearestAppointmentPractitioner={
      Object {
        "humanName": Object {
          "familyName": "Ling",
          "givenNames": Array [],
          "prefixes": Array [
            "Dr.",
          ],
        },
        "id": "1",
        "role": "MA",
      }
    }
    title="Physical Therapy"
  />
</FadedScroll>
`;
